tf2_gnn/utils/polynomial_warmup_and_decay_schedule.py

```diff
@@ -30,39 +30,39 @@ def __init__(
         It is computed as:
         ```python
         def scheduled_learning_rate(step):
-        if step <= warmup_steps:
-            step = warmup_steps - min(step, warmup_steps)
-            return ((learning_rate - initial_learning_rate) *
-                    (step / warmup_steps)**power
-                ) + initial_learning_rate
-        else:
-            step = min(step - warmup_steps, decay_steps)
-            return ((learning_rate - final_learning_rate) *
-                    (1 - step / decay_steps)**power
-                ) + final_learning_rate
+            if step <= warmup_steps:
+                return ((learning_rate - initial_learning_rate) *
+                        (step / warmup_steps)**power
+                    ) + initial_learning_rate
+            else:
+                step = min(step - warmup_steps, decay_steps)
+                return ((learning_rate - final_learning_rate) *
+                        (1 - step / decay_steps)**power
+                    ) + final_learning_rate
         ```
 
         You can pass this schedule directly into a `tf.keras.optimizers.Optimizer`
         as the learning rate.
 
         Args:
-        learning_rate: A scalar `float32` or `float64` `Tensor` or a
-            Python number.  The peak learning rate.
-        warmup_steps: A scalar `int32` or `int64` `Tensor` or a Python number.
-            Must be positive.  See the warmup computation above.
-        decay_steps: A scalar `int32` or `int64` `Tensor` or a Python number.
-            Must be positive.  See the decay computation above.
-        decay_steps: A scalar `int32` or `int64` `Tensor` or a Python number.
-            Must be positive.  See the decay computation above.
-        final_learning_rate: A scalar `float32` or `float64` `Tensor` or a
-            Python number.  The minimal end learning rate.
-        power: A scalar `float32` or `float64` `Tensor` or a
-            Python number.  The power of the polynomial. Defaults to linear, 1.0.
-        name: String.  Optional name of the operation.
+            learning_rate: A scalar `float32` or `float64` `Tensor` or a
+                Python number.  The peak learning rate.
+            warmup_steps: A scalar `int32` or `int64` `Tensor` or a Python number.
+                Must be positive.  See the warmup computation above.
+            decay_steps: A scalar `int32` or `int64` `Tensor` or a Python number.
+                Must be positive.  See the decay computation above.
+            initial_learning_rate: A scalar `float32` or `float64` `Tensor` or a
+                Python number.  The minimal starting learning rate.
```

Comment:
Does the word "minimal" make sense here? It does for `final_learning_rate`, since we may not anneal to the final value, but for the initial value we will use that at `step = 0`?

Reply:
Thanks, you're right that the wording is a bit confusing here. I've improved and clarified now.